### ISSUE TITLE
Filename not shown

### ISSUE THREAD

**Issue by furchtlos76** [pulsejet/memories#380]
Is it somehow possible to let the filenames shown in shared folders? Would like to use this for Photography things, like sharing photos to friends or customers, able to tell me the filename of the photos they would like to get edited for example.

**Comment by gsedej** [comment]
I would also like Memories to show filenames - I would like to share videos (using transcoding features) with option to show filename. Eg. recording of song.

**Comment by gsedej** [comment]
Thank you for replying. It does show, which is nice. But it would be even better if there was an option to also show filenames.

Example - screenshot below, you can't say by thumbnail what you want to listen/watch.
_(I do understand that this is a bit off scope of this project, but it's really nice feature to quickly share videos, without manually encoding them to lower resolution)_

![slika](https://user-images.githubusercontent.com/6927198/226748985-b48b19c0-9982-42af-8208-b3b37c57c0e9.png)

**Comment by furchtlos76** [comment]
> Note: you can already see the filename in the sidebar.

Oh? I Dont see a sidebar when open a link that i shared to a customer or friend!? Do i have something configured wrong?

**Comment by gsedej** [comment]
@furchtlos76 when you click on file, you can click on "i" and then it shows. _(but yes, at least 2 clicks)_

![slika](https://user-images.githubusercontent.com/6927198/226754167-f7de3e2f-2d0e-4314-9af2-77817a25642e.png)

**Comment by TommasoCostanzo** [comment]
I would also really love directly seeing the filenames. Also, related to similar use cases described above, I would love to be able to order the file in a folder of Memories by filename. In any case, thank you @pulsejet for the awesome app!

**Comment by pulsejet** [comment]
I'm concerned this is largely out of scope of memories. When we add filenames and sorting with filenames, it's essentially duplicating the functions of the files app, which doesn't make a lot of sense. The next thing that'll come up is the ability to copy, rename, share files etc etc; implementing these again is just a waste of time (not to mention keeping up with newer features there).

So the real "solution" for these cases is to interface with files itself. For example if the primary use case here is sharing videos with transcoding, it's perfectly possible to implement some adapter in the Nextcloud viewer to do this too; then all that's needed is to share the folder and open it in the files app.

tl;dr there has to be a clear separation of concerns. Memories is *not* a files app replacement for media, it's specifically a photo app. Ideally, the user should never see any "files" at all; just the photos.

**Comment by TommasoCostanzo** [comment]
> I'm concerned this is largely out of scope of memories. When we add filenames and sorting with filenames, it's essentially duplicating the functions of the files app, which doesn't make a lot of sense. The next thing that'll come up is the ability to copy, rename, share files etc etc; implementing these again is just a waste of time (not to mention keeping up with newer features there).
> 
> So the real "solution" for these cases is to interface with files itself. For example if the primary use case here is sharing videos with transcoding, it's perfectly possible to implement some adapter in the Nextcloud viewer to do this too; then all that's needed is to share the folder and open it in the files app.
> 
> tl;dr there has to be a clear separation of concerns. Memories is _not_ a files app replacement for media, it's specifically a photo app. Ideally, the user should never see any "files" at all; just the photos.

Totally agree with that! The attracting part of Memories is exactly the video transcoding. Do you see it feasible that Files will use Memories transcoding when opening a video? Because I am afraid the Files developers would not like to deal with ffmpeg dependencies directly in the Files app.

**Comment by furchtlos76** [comment]
> I'm concerned this is largely out of scope of memories. When we add filenames and sorting with filenames, it's essentially duplicating the functions of the files app, which doesn't make a lot of sense. The next thing that'll come up is the ability to copy, rename, share files etc etc; implementing these again is just a waste of time (not to mention keeping up with newer features there).
> 
> So the real "solution" for these cases is to interface with files itself. For example if the primary use case here is sharing videos with transcoding, it's perfectly possible to implement some adapter in the Nextcloud viewer to do this too; then all that's needed is to share the folder and open it in the files app.
> 
> tl;dr there has to be a clear separation of concerns. Memories is _not_ a files app replacement for media, it's specifically a photo app. Ideally, the user should never see any "files" at all; just the photos.

Thanks for that statement, well memories is a photo app, sure, but imagine people would love to use thise great app also for professional things, like sharing photos to customers, as the shared albums/photos look so much better than with the original photo/files app because of the different thumbnails. so it would be really nice to have some option to at least show the filenames, when you hover or click on a photo open in lightbox or maybe be able to let viewers select pictures somehow. for me as photographer it woulde be so nice to use memories as a presentation tool for my pictures, lettiong customers beeing able to select the pictures they would like to buy, edited or whatever - but that would need to be easy able to copy/paste the filenames out of an album. Thanks so much anyway!

**Comment by pulsejet** [comment]
> me as photographer it woulde be so nice to use memories as a presentation tool for my pictures, lettiong customers beeing able to select the pictures they would like to buy, edited or whatever - but that would need to be easy able to copy/paste the filenames out of an album.

Precisely my point. Using memories for a purpose that it wasn't built for will lead to needing more features; the cycle doesn't end ;)

Of course, the general direction of the project isn't necessarily rigid; which is why we should keep this issue open. But for now, adding any "files" functionality would just lead to more people attempting to use memories for what it isn't being (currently) built for, leading to a worse experience on average. Instead of adding these features in memories, we really should be adding functionality to the files/viewer apps; that is the sustainable solution here.

**Comment by cmaloney3030** [comment]
Could you maybe add a tooltip when you hover over the photo/video that displays the file name?  Wouldn't require a redesign.  Seems like it could be simple enough :)
